samples/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -10,78 +10,367 @@
+	event->arg = tmp_peer_id;
+
+	EVENT_SUBMIT(event);
+}
+
+static void select_confirm(void)
+{
+	LOG_INF("Select peer");
+
+	struct ble_peer_operation_event *event = new_ble_peer_operation_event();
+
+	if (tmp_peer_id != cur_peer_id) {
+		cur_peer_id = tmp_peer_id;
+
+		event->op = PEER_OPERATION_SELECTED;
+		event->arg = cur_peer_id;
+	} else {
+		event->op = PEER_OPERATION_CANCEL;
+		event->arg = 0;
+	}
+
+	EVENT_SUBMIT(event);
+}
+
+static void erase_start(void)
+{
+	LOG_INF("Start peer erase");
+
+	struct ble_peer_operation_event *event = new_ble_peer_operation_event();
+
+	event->op = PEER_OPERATION_ERASE;
+	event->arg = 0;
+
+	EVENT_SUBMIT(event);
+}
+
+static void erase_confirm(void)
+{
+	LOG_INF("Erase peer");
+
+	int err = remove_peers();
+	if (!err) {
+		struct ble_peer_operation_event *event =
+			new_ble_peer_operation_event();
+
+		event->op = PEER_OPERATION_ERASED;
+		event->arg = 0;
+
+		EVENT_SUBMIT(event);
 	} else {
 		module_set_state(MODULE_STATE_READY);
 	}
 }
 
-#if CONFIG_SHELL
-void peer_info_show(const struct bt_bond_info *info, void *user_data)
+static enum click get_click(u32_t time_diff)
 {
-	const struct shell *shell = (const struct shell *)user_data;
-	char addr[BT_ADDR_LE_STR_LEN];
+	if (time_diff < SHORT_CLICK_MAX) {
+		return CLICK_SHORT;
+	}
+	return CLICK_NONE;
+}
 
-	bt_addr_le_to_str(&info->addr, addr, sizeof(addr));
-	shell_print(shell, "\t%s", addr);
+static void handle_click(enum click click)
+{
+	if (click != CLICK_NONE) {
+		for (size_t i = 0; i < ARRAY_SIZE(state_switch); i++) {
+			if ((state_switch[i].state == state) &&
+			    (state_switch[i].click == click)) {
+				if (NULL != state_switch[i].func) {
+					state_switch[i].func();
+				}
+				state = state_switch[i].next_state;
+				if (state != STATE_IDLE) {
+					k_delayed_work_submit(&timeout,
+							OPERATION_TIMEOUT);
+				} else {
+					k_delayed_work_cancel(&timeout);
+				}
+				return;
+			}
+		}
+	}
+
+	cancel_operation();
 }
 
-static int cmd_bonds_show(const struct shell *shell, size_t argc, char **argv)
+static void timeout_handler(struct k_work *work)
 {
-	shell_print(shell, "Bonded peers:");
-	bt_foreach_bond(BT_ID_DEFAULT, peer_info_show, (void *)shell);
+	__ASSERT_NO_MSG(state != STATE_DISABLED);
 
-	return 0;
+	cancel_operation();
 }
 
-static int cmd_bonds_remove(const struct shell *shell, size_t argc, char **argv)
+static void single_click_handler(struct k_work *work)
 {
-	int err = bonds_remove();
+	handle_click(CLICK_SHORT);
+	single_click.work_q = NULL;
+}
 
-	return err;
+static void long_click_handler(struct k_work *work)
+{
+	timestamp = 0;
+	handle_click(CLICK_LONG);
+	single_click.work_q = NULL;
 }
 
-SHELL_STATIC_SUBCMD_SET_CREATE(sub_bonds,
-	SHELL_CMD_ARG(show, NULL, "Show bonds", cmd_bonds_show, 0, 0),
-	SHELL_CMD_ARG(remove, NULL, "Remove all bonded devices",
-		      cmd_bonds_remove, 0, 0),
-	SHELL_SUBCMD_SET_END
-);
+static void load_identities(void)
+{
+	bt_addr_le_t addrs[CONFIG_BT_ID_MAX];
+	size_t count;
```

Comment:
`count` Should be initialized to the `addrs` array size before calling `bt_id_get`

Reply:
I guess it does not matter really as documentation states
`the number of elements in the @a addrs array should be CONFIG_BT_ID_MAX`
but later it states `count Should be initialized to the array size` so I will set it

Reply:
Fixed